src/libraries/System.Net.Requests/tests/HttpWebRequestTest.cs

```diff
@@ -2077,6 +2077,68 @@ public async Task SendHttpGetRequest_WithCachePolicyBypassCache_DoNotAddHeaders(
             });
         }
 
+        [Fact]
+        public async Task SendHttpPostRequest_BufferingDisabled_ConnectionShouldStartWithRequestStream()
+        {
+            await LoopbackServer.CreateClientAndServerAsync(
+                async (uri) =>
+                {
+                    HttpWebRequest request = WebRequest.CreateHttp(uri);
+                    request.Method = "POST";
+                    request.AllowWriteStreamBuffering = false;
+                    var stream = await request.GetRequestStreamAsync();
+                    await Assert.ThrowsAnyAsync<Exception>(() => request.GetResponseAsync());
+                },
+                async (server) => 
+                {
+                    await server.AcceptConnectionAsync(_ =>
+                    {
+                        return Task.CompletedTask;
+                    });
+                }
+            );
+        }
+
+        [Theory]
+        [InlineData(true)]
+        [InlineData(false)]
+        public async Task SendHttpPostRequest_WhenBufferingChanges_Success(bool buffering)
+        {
+            const string text = "Hello World!!!!\n";
+            await LoopbackServer.CreateClientAndServerAsync(
+                async (uri) =>
+                {
+                    int size = text.Length * 128 * 1024;
+                    HttpWebRequest request = WebRequest.CreateHttp(uri);
+                    request.ContentType = "application/text";
+                    if (!buffering)
+                    {
+                        request.ContentLength = size;
```

Comment:
This shouldn't be necessary.

Reply:
It should actually, if we're not buffering we should either set `Content-Length` or we should mark it as `SendChunked`.

Reply:
No, that should do the web request (in this case `HttpContent`) on it's own.
If this wasn't necessary before, it cannot be required now, it will break users.

See this code path:
https://github.com/dotnet/runtime/blob/a3f74163d65fbbd81b9ed946e9d74f080be1dc70/src/libraries/System.Net.Http/src/System/Net/Http/Headers/HttpContentHeaders.cs#L34-L65

It calls `TryComputeLength` on the content:
https://github.com/dotnet/runtime/blob/a3f74163d65fbbd81b9ed946e9d74f080be1dc70/src/libraries/System.Net.Http/src/System/Net/Http/StreamContent.cs#L77-L89

So that's where the decision about ContentLength vs Chunked should be made. Because this all gets called from `SocketsHttpHandler` and it does the right thing:
https://github.com/dotnet/runtime/blob/a3f74163d65fbbd81b9ed946e9d74f080be1dc70/src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/SocketsHttpHandler.cs#L631-L635

Reply:
It's not necessary anymore, if we don't provide `Content-Length` `HttpWebRequest` will use `Transfer-Encoding: chunked`.